nemo/collections/tts/models/fastpitch.py

```diff
@@ -195,13 +195,21 @@ def parser(self):
         return self._parser
 
     def parse(self, str_input: str, normalize=True) -> torch.tensor:
+        if self.training:
+            logging.warning("parse() is meant to be called in eval mode.")
         if str_input[-1] not in [".", "!", "?"]:
             str_input = str_input + "."
 
         if normalize and self.text_normalizer_call is not None:
             str_input = self.text_normalizer_call(str_input, **self.text_normalizer_call_kwargs)
 
-        tokens = self.parser(str_input)
+        if self.learn_alignment:
+            # Disable mixed g2p representation
+            with self.vocab.set_phone_prob():
```

Comment:
Good idea. Could we set explicitly `prob=1.0` or rename it to `self.vocab.disable_mixed_representation()`, because I don't have idea in which case we can run this context manager except this one.

Reply:
Made prob required.